examples/recording_service/leveldb-integration/cpp/README.md

```diff
@@ -0,0 +1,1077 @@
+database with the stream name, taken from the `StreamInfo` object
+passed as a parameter, and the domain ID, provided as a parameter
+too.
+
+Another key aspect of the constructor is to provide a sensible
+initialization of the buffers (byte vectors) that will be used
+to serialize the samples stored in the database. This will
+minimize the number of memory allocations performed by the
+vectors.
+
+For the key buffer's initialization size, we will use the key
+type's max sample serialized size:
+
+```cpp
+const StructType& key_type = rti::topic::dynamic_type<UserDataKey>::get();
+key_buffer_.resize(key_type.cdr_serialized_sample_max_size());
+```
+
+In the case of the value buffer, the serialized sample's max size
+not only depends on the size of type `UserDataValue`, but mostly
+on the size of the topic's type's sample max serialized size. Hence,
+we first get the type from the `StreamInfo` object, which is
+represented as a native `DDS_TypeCode` pointer object. With it, we
+do the following:
+
+- If the type is unbounded, we initialize the buffer to 8 KB.
+
+- Else, we get the max serialized sample from the type and we
+  resize the buffer to that size. 
+  
+This is the code snippet for the above calculations:
+
+```cpp
+DDS_TypeCode *type = static_cast<DDS_TypeCode *>(
+        stream_info.type_info().type_representation());
+DDS_ExceptionCode_t ex = DDS_NO_EXCEPTION_CODE;
+if (DDS_TypeCode_is_unbounded(type, DDS_BOOLEAN_FALSE, &ex)
+        && ex != DDS_NO_EXCEPTION_CODE) {
+    value_buffer_.resize(8192);
+} else {
+    DDS_UnsignedLong size = DDS_TypeCode_get_cdr_serialized_sample_max_size(
+            type,
+            DDS_AUTO_DATA_REPRESENTATION,
+            &ex);
+    if (ex != DDS_NO_EXCEPTION_CODE) {
+        // ... handle error
+    }
+    value_buffer_.resize((std::vector<char>::size_type) size);
+}
+```
+
+**Store method**
+
+The `store()` method will be called by *Recorder* when samples of the
+topic are available to be stored into the database. So the main goal
+of this method is:
+
+- To iterate over all samples, passed in as a vector of pointers to
+  `dds::core::xtypes::DynamicData`, and all sample info objects.
+  
+    - For every sample, prepare a **key** that is unique: use the
+      *original publication virtual GUID* and the *original publication
+      virtual sequence number*. Store the reception timestamp
+      alongside, so we can quickly implement time-series ordering
+      of the samples (see class `UserDataKeyComparator`). All these
+      fields are available in the `SampleInfo` object.
+      
+    - For every sample, prepare a **value** entry: get the valid data
+      flag and store it in the value object. If the sample is valid,
+      serialize the sample into CDR. Here is the code snippet:
+        ```cpp
+        UserDataValue value;
+        value.valid_data(sample_info.valid());
+        if (sample_info.valid()) {
+            const dds::core::xtypes::DynamicData& sample = *(sample_seq[i]);
```

Comment:
why this line? why not passing *sample_seq[i] directly to_cdr_buffer()?

Reply:
For clarity, for the users reading the code and the doc.